### ISSUE TITLE
i2c watch dog

### ISSUE THREAD

**Issue by patjazz** [somhi/ESP_SSD1306#1]
i use it on an esp12 IIC (gpio 4,5) 
just a problem of watch dog on testdrawlines , testdraw circles (when time to draw increase): 

 ets Jan  8 2013,rst cause:4, boot mode:(3,7)

wdt reset
load 0x40100000, len 28780, room 16 
tail 12
chksum 0x55
ho 0 tail 12 room 4
load 0x3ffe8000, len 2372, room 12 
tail 8
chksum 0xff
load 0x3ffe8950, len 1784, room 0 
tail 8
chksum 0xa4
csum 0xa4

just you have to add "  yield() " somewhere ???

**Comment by RichardNobel** [comment]
Hi @patjazz 

I read you are using @somhi 's modified adafruit library with an IIC connection to your display module.

Can you show/explain to me, how you connected the pins?

I got a 128x64 OLED Display Module from eBay. It has 6 pins:
D/C
RST
SDA
SCL
VCC
GND

So I assume it is set up for IIC/I2C communication. But the example Arduino Sketch code was, according to the comments, written for hardware SPI communication.

**Comment by somhi** [comment]
I suppose we can now close this issue with @sheaivey's contributions to the code.

**Comment by sheaivey** [comment]
Great, I did have the exact issue @patjazz had with I2C.
Adding the yield() to the display function caused everything to work work as expected.

I tested on the ESP-12E and Arduino v1.6.5 with ESP v2.0.0. Also did a few tests with wifi network to make sure the ESP wifi functions still worked.

Glad I could help. Your library was the first library I found when searching google for ESP-12E OLED I2C. Thanks for doing all the heavy lifting for this library!
